### PR TITLE
Extend feature set of dynamic functions

### PR DESCRIPTION
**Pull request by Konti08**
- Add: Math and constant functionalities
- Enhance: Syntax to enaable new functionalities

### CODE REVIEW

File: dynamic_functions/constantsFunction.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+def integer(session_id, integer):
+    return int(integer)
+
+
+def float(session_id, float):
+    return float(float)
+
+
+def boolean(session_id, boolean):
+    return bool(boolean)
```
Comment (Empiire00) [inline dynamic_functions/constantsFunction.py:14]: why?

File: dynamic_functions/constantsFunction.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+def integer(session_id, integer):
+    return int(integer)
+
+
+def float(session_id, float):
+    return float(float)
+
+
+def boolean(session_id, boolean):
```
Comment (Empiire00) [inline dynamic_functions/constantsFunction.py:13]: session_id not used

File: utils/actionAssembler.py [hunk 3]
```diff
@@ -170,20 +173,68 @@ def replace_icons(actions_json):
 
 def replace_dynamic_values(value, session_id):
     if type(value) is str:
-        if value.startswith('$dynamic:'):
-            value = evaluate_dynamic_values(value.split(':')[-1], session_id)
+        if value.startswith(functions.DYNAMIC_START_SEQUENCE):
+            value = evaluate_dynamic_values(value.split(':')[1:], session_id)
```
Comment (Empiire00) [inline utils/actionAssembler.py:177]: ~~value.split(':')[-1]~~
value .split(functions.DYNAMIC_SEPARATOR)

File: utils/actionAssembler.py [hunk 4]
```diff
@@ -170,20 +173,68 @@ def replace_icons(actions_json):
...
-        for index, item in enumerate(value):
-            if type(item) is str:
-                if item.startswith('$dynamic:'):
-                    value[index] = evaluate_dynamic_values(item.split(':')[-1], session_id)
-
+        result_list = deepcopy(value)
+        for index, value in enumerate(value):
+            if type(value) is str:
+                if value.startswith(functions.DYNAMIC_START_SEQUENCE):
+                    values = value.split(':')[1:]
```
Comment (Empiire00) [inline utils/actionAssembler.py:183]: ~~value.split(':')[1:]~~ \n value.split(functions.DYNAMIC_SEPARATOR)[1:]

File: utils/actionAssembler.py [hunk 5]
```diff
@@ -170,20 +173,68 @@ def replace_icons(actions_json):
...
+        if not (values[number_of_arguments].startswith(functions.DYNAMIC_TERMINAL_INDICATOR) and \
+                values[number_of_arguments].endswith(functions.DYNAMIC_TERMINAL_INDICATOR)):
+            return False
+    return True
+
+
+def strip_terminal_values(number_of_arguments, values):
+    stripped_values = []
+    for value in values[:number_of_arguments]:
+        stripped_values.append(value[1:-1])
```
Comment (Empiire00) [inline utils/actionAssembler.py:235]: May not work with terminal symbols with len(functions.DYNAMIC_TERMINAL_INDICATOR) >1

File: dynamic_functions/constantsFunction.py (hunk 1 above)
Comment (Konti08) [reply dynamic_functions/constantsFunction.py:14]: These are the function that are actually called by the action assembler when evaluating the dynamic values.

File: dynamic_functions/constantsFunction.py (hunk 2 above)
Comment (Konti08) [reply dynamic_functions/constantsFunction.py:13]: > session_id not used

Yes, thats true but the session id must be included from all dynamic function. Due to the way how these functions are called.

File: utils/actionAssembler.py (hunk 3 above)
Comment (Konti08) [reply utils/actionAssembler.py:177]: > ~value.split(':')[1:]
> ~value.split(functions.DYNAMIC_SEPARATOR)[]

Use the dynamic start sequence defined in the module
> ~DYNAMIC_START_SEQUENCE = '$dynamic'

Comment (Konti08) [reply utils/actionAssembler.py]: > ~value.split(':')[1:]~ value.split(functions.DYNAMIC_SEPARATOR)[]

Fixed

File: utils/actionAssembler.py (hunk 4 above)
Comment (Konti08) [reply utils/actionAssembler.py]: Fixed

File: dynamic_functions/constantsFunction.py (hunk 1 above)
Comment (Empiire00) [reply dynamic_functions/constantsFunction.py:14]: aight

**Pull request merged**